### PR TITLE
[SYCLomatic] Support migration of 43 half related math APIs.

### PR DESCRIPTION
**Pull request by tangjj11**
Support migration of 43 new half related math APIs, and change migration
of 1 half related math API.

Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/lib/DPCT/APINamesMath.inc [hunk 1]
```diff
@@ -514,44 +513,44 @@ ENTRY_TYPECAST("__ull2float_rz")
 
 
 // Half Arithmetic Functions
-ENTRY_UNSUPPORTED("__hadd_sat")
-ENTRY_UNSUPPORTED("__hfma_sat")
-ENTRY_UNSUPPORTED("__hmul_sat")
-ENTRY_UNSUPPORTED("__hsub_sat")
+ENTRY_RENAMED("__hadd_sat", MapNames::getSyclMathNamespace() + "hadd_sat")
```
Comment (zhiweij1) [inline clang/lib/DPCT/APINamesMath.inc]: MapNames::getSyclMathNamespace() or MapNames::getClNamespace() + "ext::intel::math::" ?

File: clang/lib/DPCT/APINamesMath.inc [hunk 2]
```diff
@@ -514,44 +513,44 @@ ENTRY_TYPECAST("__ull2float_rz")
...
+ENTRY_RENAMED("__hfma_sat", MapNames::getSyclMathNamespace() + "hfma_sat")
```
Comment (zhiweij1) [inline clang/lib/DPCT/APINamesMath.inc]: Do we need an option to control this migration?

File: clang/test/dpct/helper_files_ref/include/util.hpp [hunk 3]
```diff
@@ -106,7 +106,7 @@ inline double cast_ints_to_double(int high32, int low32) {
 /// \param [in] a The array
 /// \param [in] len Length of the array
 /// \returns The computed fast_length
-inline float fast_length(const float *a, int len) {
+[[deprecated]] inline float fast_length(const float *a, int len) {
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/util.hpp]: I think we can keep it.

**Review by yihwang-nv — APPROVED** [review]

**Review by zhiweij1 — APPROVED** [review]

File: clang/test/dpct/cuda-math-intrinsics.cu [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// RUN: dpct --format-range=none -out-root %T/cuda-math-intrinsics %s --cuda-include-path="%cuda-path/include" -- -x cuda --cuda-host-only --std=c++14
+// RUN: dpct --format-range=none --use-dpcpp-extensions=intel_device_math -out-root %T/cuda-math-intrinsics %s --cuda-include-path="%cuda-path/include" -- -x cuda --cuda-host-only --std=c++14
```
Comment (ShengchenJ) [inline clang/test/dpct/cuda-math-intrinsics.cu]: How about keeping the default migration lit case?

Comment (tangjj11) [reply clang/test/dpct/cuda-math-intrinsics.cu]: fixed

**Review by ShengchenJ — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**